### ISSUE TITLE
ArgumentNullException with SQL Sever provider on recently released package version

### ISSUE THREAD

**Issue by gandhis1** [borisdj/EFCore.BulkExtensions#504]
I don't know exactly which version this started, but I downgraded back to 3.3.1 and everything is working again. This is only on the SQL Sever provider. Below is the stack trace, I did not get a chance to step through the debugger to see what is going on, but hopefully this gives something to go off of:

```
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at EFCore.BulkExtensions.TableInfo.CheckToSetIdentityForPreserveOrder[T](IList`1 entities, Boolean reset)
   at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.InsertAsync[T](DbContext context, Type type, IList`1 entities, TableInfo tableInfo, Action`1 progress, CancellationToken cancellationToken)
   at EFCore.BulkExtensions.SqlBulkOperation.InsertAsync[T](DbContext context, Type type, IList`1 entities, TableInfo tableInfo, Action`1 progress, CancellationToken cancellationToken)
   at EFCore.BulkExtensions.SqlBulkOperation.InsertAsync[T](DbContext context, IList`1 entities, TableInfo tableInfo, Action`1 progress, CancellationToken cancellationToken)
   at EFCore.BulkExtensions.DbContextBulkTransaction.ExecuteAsync[T](DbContext context, IList`1 entities, OperationType operationType, BulkConfig bulkConfig, Action`1 progress, CancellationToken cancellationToken)
```

**Comment by borisdj** [comment]
Could you load lib. source into your project and debug it to find what exactly is null in this method.

**Comment by gandhis1** [comment]
It's failing on accessing the `PrimaryKeyPropertyColumnNameDict` and calling `.Count()`. My entity type does not have a primary key (which should not be required for a `BulkInsert()` and hasn't been historically). Looks like this happend in 98d7565 when initializing `doSetIdentityColumnsForInsertOrder`.

**Comment by borisdj** [comment]
Thx for debug info.
Fixed with 3.5.3
